### PR TITLE
Normalize spell check score

### LINKED ISSUE #1390: FT.SPELLCHECK confidence scores extremely low for unknown reasons
I have been using redisearch for various datasets (& also seen a few examples of ft.spellcheck in github issues) and the confidence scores for the spellcheck suggestions are always ridiculously low, in fact that they don't make any sense whatsoever, even though the suggestions are really good.

```
redis-cli "FT.SPELLCHECK" "idx1" "Islad" "DISTANCE" "2" 
1) 1) "TERM"
   2) "islad"
   3)   1) 1) "0.0066979382934979018"
           2) "island"
        2) 1) "0.0031479132246004962"
           2) "+island"
        3) 1) "0.0030284574047466021"
           2) "islands"
        4) 1) "0.00087488769470457401"
           2) "+islam"
        5) 1) "0.00064270596034066781"
           2) "isaac"
        6) 1) "0.00062924614965290517"
           2) "isle"
        7) 1) "0.00049801299544721905"
           2) "island."
        8) 1) "0.00049464804277527831"
           2) "islam"
        9) 1) "0.00022713430535599517"
           2) "inland"
       10) 1) "0.0002187219236761435"
           2) "isles"
```

(via https://github.com/RediSearch/RediSearch/issues/969)

This is approx what I get for 99.9% of cases.
However https://redislabs.com/blog/redisearch-1-4-phonetics-spell-check/ shows scores of 5 and 7, never had that happen.

Additionally, I think it would be nice (more intuitive) if the scores were some normalized value between 0-1, where 0.6 means 60% confidence.

Could you check why this happens or update the docs explaining that these extremely low values are normal and how they can be interpreted?

### PR DESCRIPTION
**Pull request by ashtul**
fix #1390 
Currently, the score is divided by the number of documents in the index. Normalized to the max count of found suggestions.

### CODE REVIEW

File: src/spell_check.c [hunk 1]
```diff
@@ -173,6 +173,14 @@ RS_Suggestion **spellCheck_GetSuggestions(RS_Suggestions *s) {
   return ret;
 }
 
+static double spellCheck_GetMaxScore(RS_Suggestion **suggestions) {
```
Comment (rafie) [inline src/spell_check.c:176]: `RS_Suggestion **` -> `arrayof(RS_Suggestion *) *`

File: src/spell_check.c [hunk 2]
```diff
@@ -181,13 +189,14 @@ void SpellCheck_SendReplyOnTerm(RedisModuleCtx *ctx, char *term, size_t len, RS_
   RedisModule_ReplyWithStringBuffer(ctx, term, len);
 
   RS_Suggestion **suggestions = spellCheck_GetSuggestions(s);
```
Comment (rafie) [inline src/spell_check.c:191]: `RS_Suggestion **` -> `arrayof(RS_Suggestion *) *`

File: src/spell_check.c [hunk 3]
```diff
@@ -173,21 +173,30 @@ RS_Suggestion **spellCheck_GetSuggestions(RS_Suggestions *s) {
   return ret;
 }
 
+static double spellCheck_GetMaxScore(arrayof(RS_Suggestion) *suggestions) {
```
Comment (rafie) [inline src/spell_check.c:176]: `arrayof(X)` is "array of objects of type X", which is actually a X*.
Here X=RS_Suggestion* (this is an array of pointers), so it should be `arrayof(RS_Suggestion*) suggestions` (which maps to `RS_Suggestions **`, just like the original).

**Comment by ashtul** [comment]
might be problematic on a cluster since MaxScore is per shard

**Pull request closed without merging**